File: vdp-php/tests/vta/ManagePortfoliosTest.php
<?php

namespace Vdp;

class ManagePortfoliosTest extends \PHPUnit_Framework_TestCase {
	
	public function setUp() {
		$this->conf = parse_ini_file ( "configuration.ini", true );
		$this->abstractVisaAPIClient = new AbstractVisaAPIClient;
	}
	
	public function testPortfolios() {
		$baseUrl = "vta/";
		$resourcePath = "v3/communities/".$this->conf['VDP'] ['vtaCommunityCode']."/portfolios";
		$statusCode = $this->abstractVisaAPIClient->doMutualAuthCall( 'get', $baseUrl.$resourcePath, 'Get Portfolio Details Test', '', array("ServiceId: ".$this->conf['VDP'] ['vtaServiceId'] ));
		$this->assertEquals($statusCode, "200");
	}
}

File: vdp-c-sharp/vdp-c-sharp/VDPApiTest/vta/ManagePortfoliosTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Configuration;
using System.Collections.Generic;

namespace Vdp
{
    [TestClass]
    public class ManagePortfoliosTest
    {
        private AbstractVisaAPIClient abstractVisaAPIClient;

        public ManagePortfoliosTest()
        {
            abstractVisaAPIClient = new AbstractVisaAPIClient();
        }

        [TestMethod]
        public void TestPortfolios()
        {
            string baseUri = "vta/";
            string resourcePath = "v3/communities/" + ConfigurationManager.AppSettings["vtaCommunityCode"] + "/portfolios";
            Dictionary<string, string> headers = new Dictionary<string, string>();
            headers.Add("ServiceId", ConfigurationManager.AppSettings["vtaServiceId"]);
            string status = abstractVisaAPIClient.DoMutualAuthCall(baseUri + resourcePath, "GET", "Get Portfolio Details Test", "", headers);
            Assert.AreEqual(status, "OK");
        }
    }
}
